lib/services/database.go

```diff
@@ -200,6 +195,45 @@ func ValidateDatabase(db types.Database) error {
 	return nil
 }
 
+// validateMongoDB validates MongoDB URIs with "mongodb" schemes.
+func validateMongoDB(db types.Database) error {
+	connString, err := connstring.ParseAndValidate(db.GetURI())
+	// connstring.ParseAndValidate requires DNS resolution on TXT/SRV records
+	// for a full validation for "mongodb+srv" URIs. We will try to skip the
+	// DNS errors here by replacing the scheme and then ParseAndValidate again
+	// to validate as much as we can.
+	if isDNSError(err) {
+		logrus.Warnf("MongoDB database %q (connection string %q) failed validation with DNS error: %v.", db.GetName(), db.GetURI(), err)
+
+		connString, err = connstring.ParseAndValidate(strings.Replace(
+			db.GetURI(),
+			connstring.SchemeMongoDBSRV+"://",
+			connstring.SchemeMongoDB+"://",
+			1,
+		))
+	}
+	if err != nil {
+		return trace.BadParameter("invalid MongoDB database %q connection string %q: %v", db.GetName(), db.GetURI(), err)
+	}
+
+	// Validate read preference to catch typos early.
+	if connString.ReadPreference != "" {
+		if _, err := readpref.ModeFromString(connString.ReadPreference); err != nil {
+			return trace.BadParameter("invalid MongoDB database %q read preference %q", db.GetName(), connString.ReadPreference)
+		}
+	}
+	return nil
+}
+
+func isDNSError(err error) bool {
+	if unwrapped := errors.Unwrap(err); unwrapped != nil {
+		err = unwrapped
+	}
+
+	_, ok := err.(*net.DNSError)
+	return ok
```

Comment:
nit: can we just call ? 
```suggestion
	var dnsErr *net.DNSError
	return errors.As(err, &dnsErr)
```